airbyte-integrations/connectors/source-salesforce/source_salesforce/spec.yaml

```diff
@@ -60,9 +60,21 @@ connectionSpecification:
       description: Toggle to use Bulk API (this might cause empty fields for some streams)
       default: false
       order: 6
+    stream_slice_step:
```

Comment:
Should this be hidden? How are users able to set hidden config fields? My fears are:
* Letting the customer choose that generates syncs that are very inefficient. You can check workspace:4fa4e0fe-1a88-4ce2-b8fb-4e6c6e9a4653 connection:58f37555-c64f-4de6-9762-3f688498f08a which is a good example for stripe as the user full refresh accounts with a slice range of one day. 
* The long term proposal is to automatically choose optimal STREAM_SLICE_STEP. This would mean a breaking change and the more users use this feature, the more users we risk the frustrate
* As an improvement later, I would maybe allow to configure this step by stream as it is not exactly true that all streams have the same distribution. This would probably be another breaking change and hence something that might frustrate users.

Reply:
> Should this be hidden? How are users able to set hidden config fields? 

I don't think so. Users can not see/set hidden fields in UI.

> The long term proposal is to automatically choose optimal STREAM_SLICE_STEP. This would mean a breaking change and the more users use this feature, the more users we risk the frustrate

I don't expect this feature will be used by many customers, 30D interval should be enough for most of the users.

> As an improvement later, I would maybe allow to configure this step by stream as it is not exactly true that all streams have the same distribution. This would probably be another breaking change and hence something that might frustrate users.

Agreed, per stream settings would be better. As a workaround, customer can create different connection for streams.